services/headless-lms/models/src/peer_review_queue_entries.rs

```diff
@@ -0,0 +1,120 @@
+use crate::{exercise_slide_submissions::ExerciseSlideSubmission, prelude::*};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone, Eq)]
+#[cfg_attr(feature = "ts_rs", derive(TS))]
+pub struct PeerReviewQueueEntry {
+    pub id: Uuid,
+    pub created_at: DateTime<Utc>,
+    pub updated_at: DateTime<Utc>,
+    pub deleted_at: Option<DateTime<Utc>>,
+    pub user_id: Uuid,
+    pub exercise_id: Uuid,
+    pub receiving_peer_reviews_exercise_slide_submission_id: Uuid,
+    pub peer_review_priority: i32,
+}
+
+pub async fn insert_by_exercise_slide_submission(
+    conn: &mut PgConnection,
+    exercise_slide_submission: &ExerciseSlideSubmission,
+) -> ModelResult<PeerReviewQueueEntry> {
+    let res = sqlx::query_as!(
+        PeerReviewQueueEntry,
+        "
+INSERT INTO peer_review_queue_entries (
+    user_id,
+    exercise_id,
+    receiving_peer_reviews_exercise_slide_submission_id
+  )
+VALUES ($1, $2, $3)
+RETURNING *
+        ",
+        exercise_slide_submission.user_id,
+        exercise_slide_submission.exercise_id,
+        exercise_slide_submission.id
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(res)
+}
+
+pub async fn get_by_id(conn: &mut PgConnection, id: Uuid) -> ModelResult<PeerReviewQueueEntry> {
+    let res = sqlx::query_as!(
+        PeerReviewQueueEntry,
+        "
+SELECT *
+FROM peer_review_queue_entries
+WHERE id = $1
+  AND deleted_at IS NULL
+        ",
+        id
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(res)
+}
+
+pub async fn try_to_get_by_user_and_exercise_ids(
+    conn: &mut PgConnection,
+    user_id: Uuid,
+    exercise_id: Uuid,
+) -> ModelResult<Option<PeerReviewQueueEntry>> {
+    let res = sqlx::query_as!(
+        PeerReviewQueueEntry,
+        "
+SELECT *
+FROM peer_review_queue_entries
+WHERE user_id = $1
+  AND exercise_id = $2
+  AND deleted_at IS NULL
+        ",
+        user_id,
+        exercise_id,
+    )
+    .fetch_optional(conn)
+    .await?;
+    Ok(res)
+}
+
+pub async fn try_to_get_random_other_users_peer_review_entry(
+    conn: &mut PgConnection,
+    user_id: Uuid,
+    exercise_id: Uuid,
+) -> ModelResult<Option<PeerReviewQueueEntry>> {
+    let res = sqlx::query_as!(
+        PeerReviewQueueEntry,
+        "
+SELECT *
+FROM peer_review_queue_entries
+WHERE user_id <> $1
+  AND exercise_id = $2
+  AND deleted_at IS NULL
+ORDER BY random(), peer_review_priority DESC
```

Comment:
ORDER only by peer_review_priority, and limit to queue entries that have not received enough (boolean, for performance and because we don't want updates to the limit to affect old answers)

Reply:
https://github.com/rage/quizzes/blob/6c992c3e639188f0f752953efa1273df47593e41/packages/backendv2/src/models/quiz_answer.ts#L962-L1041

Basic idea:

1. Get 10 candidates
2. shuffle them in the program
3. Then sort the ones that don't need peer reviews anymore to the bottom of the array (maybe don't need)
4. Select the top one from the array

Reply:
If the algorithm does not produce any results, then
1. Repeat the procedure but without the received enough limitation
2. If that's not enough, select any random answer